src/qr/dclogin_scheme.rs

```diff
@@ -0,0 +1,402 @@
+use std::collections::HashMap;
+
+use crate::config::Config;
+use crate::context::Context;
+use crate::provider::Socket;
+use crate::{contact, login_param::CertificateChecks};
+use anyhow::{bail, Context as _, Result};
+use num_traits::cast::ToPrimitive;
+
+use super::{Qr, DCLOGIN_SCHEME};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum LoginOptions {
+    UnsuportedVersion(u32),
+    V1 {
+        mail_pw: String,
+        imap_host: Option<String>,
+        imap_port: Option<u16>,
+        imap_username: Option<String>,
+        imap_password: Option<String>,
+        imap_security: Option<Socket>,
+        imap_certificate_checks: Option<CertificateChecks>,
+        smtp_host: Option<String>,
+        smtp_port: Option<u16>,
+        smtp_username: Option<String>,
+        smtp_password: Option<String>,
+        smtp_security: Option<Socket>,
+        smtp_certificate_checks: Option<CertificateChecks>,
```

Comment:
This could generally be a single `HashMap<Config, String>` that maps e.g. Config::Addr to the address and Config::MailPw to the password. This would save quite some boilerplate code, but of course this wouldn't make sure that no "wrong" configs are in the list ("make invalid states non-representable").

Reply:
I don't want to throw my hard manual work away to get something less safe/correct, right. I thought about it first too, but that's what the version is for, don't allow random config values that could be not yet supported by the DC scanning it, that's part of the reason it's so verbose.